Sources/swift-format/Utilities/Helpers.swift

```diff
@@ -42,16 +42,33 @@ func readSource(from fileHandle: FileHandle) -> String? {
 func processSources(
   from paths: [String], configurationPath: String?,
   diagnosticEngine: DiagnosticEngine,
-  transform: (FileHandle, String, Configuration) -> Void
+  transform: @escaping (FileHandle, String, Configuration) -> Void
 ) throws {
+  let concurrentQueue = DispatchQueue(label: "com.queue.Concurrent", qos: .userInitiated, attributes: .concurrent)
+  let group = DispatchGroup()
+  let lock = NSLock()
+  var lastError: Error?
   for path in FileIterator(paths: paths) {
-    guard let sourceFile = FileHandle(forReadingAtPath: path) else {
-      diagnosticEngine.diagnose(
-        Diagnostic.Message(.error, "Unable to create a file handle for source from \(path)."))
-      return
+    concurrentQueue.async(group: group) {
```

Comment:
You should use `DispatchQueue.concurrentPerform(iterations:execute:)`, it will simplify the code a bit and is strongly recommended over passing `async` blocks to a concurrent queue without any upper bound check for how many run concurrently. The latter can lead to thread exhaustion.

Reply:
I don't see a specific answer about this, but there was previous conversation about it https://github.com/apple/swift-format/pull/117#issuecomment-574898189 I think the file discovery is intentionally lazy so we don't know the number of iterations here

Reply:
@allevato is there a reason that the file discovery cannot be eager?

Reply:
@akyrtzi Just that at the time it was written, there was no reason for it to be, so we could avoid the startup delay of collecting everything up front if someone ran the tool with `--recursive` on a large directory structure.

Is `DispatchQueue.concurrentPerform` preferred over `OperationQueue`, which also provides control over the number of concurrent jobs?

Reply:
I pushed the `concurrentPerform` option so everyone can see what that would look like, happy to change it to whatever we want here